app/Http/Controllers/UserController.php

```diff
@@ -142,7 +147,16 @@ public function updateRewardsType(UpdateRewardsTypeRequest $request)
      */
     public function searchUser(Request $request)
     {
-        return StrippedUserResource::collection(User::where('username', 'like', '%' . $request['userSearch'] . '%')->get());
+        /** @var User */
+        $activeUser = Auth::user();
+        $blockedUsers = $activeUser->blockedUsers()->get();
+        $blockedByUsers = $activeUser->blockedByUsers()->get();
+        $exclude = $blockedUsers->concat($blockedByUsers);
+        $exclude = $exclude->map(function ($item) {
+            return $item->id;
```

Comment:
Are you sure this gets the user id? I see you've added a pivot of id, but I think you need to get this using `$item->pivot->id`. Haven't tested it, not able to run the code rn.